front_end/entrypoints/rn_fusebox/rn_fusebox.ts

```diff
@@ -222,4 +224,33 @@ UI.Toolbar.registerToolbarItem({
   },
 });
 
+class FuseboxReactNativeApplicationObserver implements
+    SDK.TargetManager.SDKModelObserver<SDK.ReactNativeApplicationModel.ReactNativeApplicationModel> {
+  constructor(targetManager: SDK.TargetManager.TargetManager) {
+    targetManager.observeModels(SDK.ReactNativeApplicationModel.ReactNativeApplicationModel, this);
+  }
+
+  modelAdded(model: SDK.ReactNativeApplicationModel.ReactNativeApplicationModel): void {
+    model.ensureEnabled();
+    model.addEventListener(SDK.ReactNativeApplicationModel.Events.MetadataUpdated, this.#handleMetadataUpdated, this);
+  }
+
+  modelRemoved(model: SDK.ReactNativeApplicationModel.ReactNativeApplicationModel): void {
+    model.removeEventListener(
+        SDK.ReactNativeApplicationModel.Events.MetadataUpdated, this.#handleMetadataUpdated, this);
+  }
+
+  #handleMetadataUpdated(
+      event: Common.EventTarget.EventTargetEvent<Protocol.ReactNativeApplication.MetadataUpdatedEvent>): void {
+    const {appDisplayName, deviceName} = event.data;
+
+    // Update window title
+    if (appDisplayName != null && deviceName != null) {
+      document.title = `${appDisplayName} (${deviceName}) - React Native DevTools`;
+    }
+  }
+}
+
+new FuseboxReactNativeApplicationObserver(SDK.TargetManager.TargetManager.instance());
```

Comment:
Prior art for this pattern under `front_end/entrypoints/`: [`ExecutionContextSelector`](https://github.com/facebookexperimental/rn-chrome-devtools-frontend/blob/main/front_end/entrypoints/main/ExecutionContextSelector.ts#L9).